### PR TITLE
Add unit tests for some boss.util modules and some other minor fixes

### LINKED ISSUE #64: Write tests for boss api and utils
Write tests for:
 * Modules and functions under `boss.api`
 * Code under `boss.util`
 * Any other code that requires tests

### PR DESCRIPTION
**Pull request by srishanbhattarai**
* Partially fixes #64
* Added simple unit tests for `boss.util` modules that don't require mocking `fabric` functions.
* Since capturing `stdout` (and possibly `stderr` in the future) became necessary, added a `-s` flag to `pytest` scripts

## Changes to existing code:
  * Improved `is_iterable` function in `boss.util` to also check for `__getitem__` as well as `__iter__` since supporters of the sequence protocol also count as iterators. More info in the [docs](https://docs.python.org/2/library/functions.html#iter). (This almost certainly won't induce anything to break)
  * Added a few missing dependencies to `requirements-dev.txt` and `setup.py`

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+)
+
+def test_halt():
+    ''' Test for boss.util.halt() '''
+    message = 'Test message'
+
+    with pytest.raises(SystemExit, match=message):
+        halt(message)
+
+def test_echo(capsys):
```
Comment (kabirbaidhya) [inline tests/test_util.py]: Inconsistent blank lines here, as [pep8](https://www.python.org/dev/peps/pep-0008/#blank-lines) requires 2 blank lines in between function definitions. You can use something like [this](https://github.com/hhatto/autopep8) too.

File: tests/test_util.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+def test_info(capsys):
+    ''' Test for boss.util.info() '''
+    message = 'Test message'
+    info(message)
+
+    out, _ = capsys.readouterr()
+
+    assert message in out
+
+def test_merge_v0():
```
Comment (kabirbaidhya) [inline tests/test_util.py]: same here.

File: tests/test_util.py (hunk 1 above)
Comment (srishanbhattarai) [reply tests/test_util.py]: Are you using an `autopep8` command to format the code? It'll be helpful to add that to `t`.

Comment (kabirbaidhya) [reply tests/test_util.py]: @srishanbhattarai Actually, I'm using a vscode plugin right now. But, I think adding a command to `t` would be pretty handy too.

**Pull request merged**